syntax = "proto3";
package distance_server;
import "google/api/annotations.proto";
import "google/protobuf/empty.proto";

message ClientInfo {
    string info = 1;
}

message ServerInfo {
    string info = 1;
}

message Distance {
    int32 id = 1;
    int64 ts = 2;
    int32 distance = 3;
}

service DistanceService {

    rpc registerClient (ClientInfo) returns (ServerInfo) {
    }

    rpc getDistances (ClientInfo) returns (stream Distance) {
        option (google.api.http) = {
			get: "/v1/distances"
		};
    }

    rpc getDistance (google.protobuf.Empty) returns (Distance) {
        option (google.api.http) = {
            get: "/v1/distance"
        };
    }
}


